### PR TITLE
Allow message when suspending relation; suspnd/resume more than 1

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

The suspend/resume relation CLI gains the ability to suspend/resume more than one relation, plus add a message to relation status (on the offering side) when suspending a relation.

As a driveby, fix exports for models with remote applications with relations. This fix allows juju dump-model to work.

## QA steps

Deploy a cmr scenario.
Suspend remote relation(s) with a message
Look at juju offers --format yaml and check the status message on the offer
Ensure that juju dump-model works with suspended relations.

### CODE REVIEW

File: cmd/juju/application/resumerelation.go [hunk 1]
```diff
@@ -47,25 +49,29 @@ func NewResumeRelationCommand() cmd.Command {
...
 		Purpose: resumeHelpSummary,
 		Doc:     resumeHelpDetails,
 	}
 }
 
 func (c *resumeRelationCommand) Init(args []string) (err error) {
 	if len(args) == 1 {
-		if c.RelationId, err = strconv.Atoi(args[0]); err != nil || c.RelationId < 0 {
-			return errors.NotValidf("relation ID %q", args[0])
+		idStr := strings.Split(args[0], ",")
```
Comment (howbazaar) [inline cmd/juju/application/resumerelation.go]: Please separate into just positional args.

File: cmd/juju/application/suspendrelation.go [hunk 2]
```diff
@@ -40,44 +44,53 @@ func NewSuspendRelationCommand() cmd.Command {
...
 		Purpose: suspendHelpSummary,
 		Doc:     suspendHelpDetails,
 	}
 }
 
 func (c *suspendRelationCommand) Init(args []string) (err error) {
 	if len(args) == 1 {
-		if c.RelationId, err = strconv.Atoi(args[0]); err != nil || c.RelationId < 0 {
-			return errors.NotValidf("relation ID %q", args[0])
+		idStr := strings.Split(args[0], ",")
```
Comment (howbazaar) [inline cmd/juju/application/suspendrelation.go]: Same here, multiple positional args.

File: cmd/juju/application/suspendrelation_test.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ func (s *SuspendRelationSuite) runSuspendRelation(c *gc.C, args ...string) error
 func (s *SuspendRelationSuite) TestSuspendRelationWrongNumberOfArguments(c *gc.C) {
 	// No arguments
 	err := s.runSuspendRelation(c)
-	c.Assert(err, gc.ErrorMatches, "no relation id specified")
+	c.Assert(err, gc.ErrorMatches, "no relation id\\(s\\) specified")
```
Comment (howbazaar) [inline cmd/juju/application/suspendrelation_test.go]: Proper plural response please. No (s).

Comment (wallyworld) [reply cmd/juju/application/suspendrelation_test.go]: this isn't a plural - it's an error message saying the user did not specify the required one (or more) relation ids. nonetheless, i've removed the brackets

File: cmd/juju/application/suspendrelation.go (hunk 2 above)
Comment (wallyworld) [reply cmd/juju/application/suspendrelation.go]: done

File: cmd/juju/application/resumerelation.go (hunk 1 above)
Comment (wallyworld) [reply cmd/juju/application/resumerelation.go]: done

File: cmd/juju/application/suspendrelation.go [hunk 4]
```diff
@@ -40,44 +44,49 @@ func NewSuspendRelationCommand() cmd.Command {
...
+	relationIds []int
+	message     string
+	newAPIFunc  func() (SetRelationSuspendedAPI, error)
 }
 
 func (c *suspendRelationCommand) Info() *cmd.Info {
 	return &cmd.Info{
 		Name:    "suspend-relation",
-		Args:    "<relation-id>",
+		Args:    "[--message <message>] <relation-id>[ <relation-id>...]",
```
Comment (howbazaar) [inline cmd/juju/application/suspendrelation.go]: Don't specify options in the Args section.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**